### PR TITLE
feat: make it possible to also check annotations/labels when matching

### LINKED ISSUE #1392: Make it possible for `GenericKubernetesResourceMatcher` consider metadata as well
More specifically annotations and labels should be checked since the rest of the fields are either irrelevant (e.g. `name`) or not under a user's control (e.g. `managedFields`).

### PR DESCRIPTION
**Pull request by metacosm**
Fixes #1392

### CODE REVIEW

**Review by csviri — CHANGES_REQUESTED** [review]
What is not clear to me in this impl, how it should be used in a real life dependent resource, that should override the `match` method? 
Should we showcase that somehow?

File: docs/documentation/dependent-resources.md [hunk 1]
```diff
@@ -169,26 +198,26 @@ instances are managed by JOSDK, an example of which can be seen below:
 ```java
 
 @ControllerConfiguration(
-        labelSelector = SELECTOR,
```
Comment (csviri) [inline docs/documentation/dependent-resources.md:172]: I don't get this, we are both using the same formatter from maven, how was this not formatted? It should fail also on build if not properly formatted

Comment (metacosm) [reply docs/documentation/dependent-resources.md:172]: I don't know but I've seen weird things going on with format… it doesn't appear as deterministic as it should be (or rather, the determinism is not obvious to me) 😓

**Review by csviri — APPROVED** [review]
LGTM

**Pull request merged**